app.js

```diff
@@ -0,0 +1,113 @@
+/*
+* app.js: Entry point for the presenter
+*
+* (C) 2015 Rackspace, Inc.
+*
+*/
+
+function normalize_url(url) {
+  if (url.slice(-1) == '/') {
+    return url.slice(0, -1);
+  }
+  return url;
+}
+
+var mapping_service_url;
+var content_service_url;
+
+// TODO: create function that long-polls and waits for an etcd key change, then calls
+// get_mapping_service_url() and get_content_service_url() to update the
+// mapping_service_url and content_service_url vars.
+
+function set_mapping_service_url() {
+  // TODO: replace process.env with etcd call
+  mapping_service_url = process.env.MAPPING_SERVICE_URL;
+}
+
+function set_content_service_url() {
+  // TODO: replace process.env with etcd call
+  content_service_url = process.env.CONTENT_SERVICE_URL;
+}
+
+set_mapping_service_url();
+set_content_service_url();
+
+var http = require('http');
+var express = require('express'),
+    exphbs  = require('express-handlebars');
+
+var app = express();
+app.engine('handlebars', exphbs());
+app.set('view engine', 'handlebars');
+
+app.get('/', function(req, presenterRes) {
+  // form route to query
+  mapping_route = normalize_url(mapping_service_url) + '/at/' + req.hostname + req.path;
+  // query mapping service for content ID
+  http.get(mapping_route, function(mappingRes) {
```

Comment:
I'm :+1: on using a library. Node land is all about having many tiny modules that do one thing well, and shipping with Docker means we shouldn't have issues keeping development and production consistent even with many of them.